packages/svelte/src/internal/client/dom/elements/attributes.js

```diff
@@ -16,12 +16,17 @@ import { LOADING_ATTR_SYMBOL } from '../../constants.js';
  */
 export function remove_input_attr_defaults(dom) {
 	if (hydrating) {
-		// using getAttribute instead of dom.value allows us to have
-		// null instead of "on" if the user didn't set a value
-		const value = dom.getAttribute('value');
-		set_attribute(dom, 'value', null);
-		set_attribute(dom, 'checked', null);
-		if (value) dom.value = value;
+		// @ts-expect-error
+		dom.__on_r = () => {
+			// TODO: do we really need to apply this for selects? the value attribute has no effect on the reset behavior
+			if (dom.tagName === 'INPUT') {
+				/** @type {HTMLInputElement} */ (dom).defaultValue = '';
+				/** @type {HTMLInputElement} */ (dom).defaultChecked = false;
+			} else {
+				dom.value = '';
+			}
```

Comment:
Alternatively this can become
```js
		const value = dom.getAttribute('value');
		set_attribute(dom, 'value', null);
		set_attribute(dom, 'checked', null);
		if (value) dom.value = value;
```
like before, because it then removes the value attribute completely. Not sure how relevant that is.